src/users/user.php

```diff
@@ -156,9 +159,17 @@ public function post()
 	 */
 	public function get()
 	{
-		$input = JFactory::getApplication()->input;
-		$id = $input->get('id', 0, 'int');
-		$xidentifier	= $input->server->get('HTTP_X_IDENTIFIER', '', 'String');
+		$input       = Factory::getApplication()->input;
+		$id          = $input->get('id', 0, 'int');
+		$xidentifier = $input->server->get('HTTP_X_IDENTIFIER', '', 'String');
+		$user        = Factory::getUser();
+
+		if (!$user->authorise('core.admin') && $id != $user->id)
```

Comment:
Should this be hardcoded to `core.admin` ?  I suggest you check for the com_users create & edit permission based on if it's a create or edit user case.